### ISSUE TITLE
RCS1171 doesn't trigger for complex variable initialisation

### ISSUE THREAD

**Issue by IanKemp** [JosefPihrt/Roslynator#114]
I requested RCS1171 in #43 and am very grateful it was added. However it only seems to fire in situations where the object initialisation is a simple one-liner, e.g. the following doesn't trigger it:

```csharp
private List<string> _listOfStrings;

public List<string> ListOfStrings
{
    get
    {
        if (_listOfStrings == null)
        {
            _listOfStrings = new List<string>
            {
               "foo",
               "bar",
            };
        }

        return _listOfStrings;
    }
}
```

I would expect the above would trigger RCS1171 which would rewrite it as follows:

```csharp
private List<string> _listOfStrings;

public List<string> ListOfStrings
{
    get
    {
        return _listOfStrings ?? (_listOfStrings = new List<string>
        {
           "foo",
           "bar",
        });
    }
}
```

**Comment by JosefPihrt** [comment]
Fixed in #195
